src/coreclr/vm/fcall.h

```diff
@@ -799,6 +799,22 @@ LPVOID __FCThrowArgument(LPVOID me, enum RuntimeExceptionKind reKind, LPCWSTR ar
 #define HELPER_METHOD_FRAME_GET_RETURN_ADDRESS()                                        \
     ( static_cast<UINT_PTR>( (__helperframe.InsureInit(NULL)), (__helperframe.MachineState()->GetRetAddr()) ) )
 
+#define EXCEPTION_METHOD_FRAME_BEGIN(funCallDepth) \
```

Comment:
I do not think this needs to be wrapped in a macro. Inlining the code to two places where it is needed would be more readable.

Reply:
Again, I've missed your comment on the fact that `FCThrow` will go away. I am not sure what the plan is though. For example, the `JIT_Div` throws uses the `FCThrow` when there is division by zero or overflow. And there are other throwing helpers like `JIT_ThrowMethodAccessException`, `JIT_ThrowFieldAccessException`. And also some throw helpers that are QCALLs now ExceptionNative_ThrowEntryPointNotFoundException that I think could also get this improvement.

Reply:
> For example, the JIT_Div throws uses the FCThrow when there is division by zero or overflow. 

We will convert them into the mix of C#, FCalls without HMF, and QCalls as needed. For example, check https://github.com/dotnet/runtime/pull/102129 that converted multiplication helpers that followed very similar pattern.

> And also some throw helpers that are QCALLs now ExceptionNative_ThrowEntryPointNotFoundException that I think could also get this improvement.

Any QCall can throw exception. If we were to improve something about QCalls, it should be a scheme that applies to all QCalls. My preference would be switch QCalls to an ordinary exception marshalling scheme (ie it is what code outside runtime needs to do to marshal exceptions between managed/unmanaged): store the exception on the unmanaged side in thread-local storage and rethrow it on the managed side.

If we care enough about improving perf of specific exception-throwing QCalls, we should just move more of them to C#. There is no point in throwing C++ exception, catching it, and rethrowing it as managed exception. We can throw the managed exception directly from managed code.